plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryArrowBufferAllocator.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.bigquery;
+
+import com.google.inject.Inject;
+import io.airlift.log.Logger;
+import org.apache.arrow.memory.AllocationListener;
+import org.apache.arrow.memory.AllocationOutcome;
+import org.apache.arrow.memory.BufferAllocator;
+import org.apache.arrow.memory.RootAllocator;
+import org.weakref.jmx.Flatten;
+import org.weakref.jmx.Managed;
+
+import static java.util.Objects.requireNonNull;
+
+public class BigQueryArrowBufferAllocator
+{
+    private static final Logger log = Logger.get(BigQueryArrowBufferAllocator.class);
+
+    private final long maximumAllocation;
+    private final BigQueryArrowAllocatorStats stats;
+    private final BufferAllocator rootAllocator;
+
+    @Inject
+    public BigQueryArrowBufferAllocator(BigQueryArrowConfig config, BigQueryArrowAllocatorStats stats)
+    {
+        this.maximumAllocation = requireNonNull(config, "config is null").getMaxAllocation().toBytes();
```

Comment:
We avoid using requireNonNull for config classes. https://github.com/trinodb/trino/pull/13940